bot/deploy.commands.js

```diff
@@ -0,0 +1,51 @@
+require('dotenv').config()
+const { REST, Routes } = require('discord.js');
+const token = process.env.token;
+const guildId = process.env.guildId;
+const clientId = process.env.clientId;
+const fs = require('node:fs');
+const path = require('node:path');
+
+console.log(token, guildId, clientId);
+
+const commands = [];
+// Grab all the command files from the commands directory you created earlier
+const foldersPath = path.join(__dirname, 'commands');
+const commandFolders = fs.readdirSync(foldersPath);
+
+for (const folder of commandFolders) {
+    // Grab all the command files from the commands directory you created earlier
+    const commandsPath = path.join(foldersPath, folder);
+    const commandFiles = fs.readdirSync(commandsPath).filter(file => file.endsWith('.js'));
+    // Grab the SlashCommandBuilder#toJSON() output of each command's data for deployment
+    for (const file of commandFiles) {
+        const filePath = path.join(commandsPath, file);
+        const command = require(filePath);
+        if ('data' in command && 'execute' in command) {
+            commands.push(command.data.toJSON());
+        } else {
+            console.log(`[WARNING] The command at ${filePath} is missing a required "data" or "execute" property.`);
+        }
+    }
+}
+
+// Construct and prepare an instance of the REST module
+const rest = new REST().setToken(token);
+
+// and deploy your commands!
+(async () => {
```

Comment:
This top level async function seems wrong to me, I know it works but I wonder if we can factor this out into a function at some point 🤔